src/models/costproxy.h

```diff
@@ -0,0 +1,43 @@
+#ifndef COSTPROXY_H
+#define COSTPROXY_H
+
+#include "treemodel.h"
+#include <QSortFilterProxyModel>
+
+template<typename Results, typename ModelImpl>
```

Comment:
why pass both types and force TreeModel below (and include it above)? Can't you just take a singular `typename Model` here, and then cast to that directly? then you can remove the `treemodel.h` include, I don't think it should be needed here